### PR TITLE
build images to cache with buildx

### PR DESCRIPTION
**Pull request by deitch**
Signed-off-by: Avi Deitcher <avi@deitcher.net>

**- What I did**

`linuxkit pkg build` utilize the linuxkit oci layout cache for builds, option to use `--docker` to place built images in the docker image cache, rather than the oci cache; parallels the `--docker` flag used for `linuxkit build`.

Includes updated documents.

**- How I did it**

Rewrote elements of `src/cmd/linuxkit/`

**- How to verify it**

CI plus push and sign tests

**- Description for the changelog**

Use linuxkit cache for package builds

**- Additional information**

This is a work in progress; I have not yet:

- [x]  updated CI
- [x]  simplified some code; there probably is too much redundancy
- [x]  completed manual tests by pushing out alternate images

I want to get peer review first.

**- Purpose**

The real purpose here is to prep the ground for cross-building. As long as the results of `lkt pkg build` are stored in a local docker engine, we have no way to cross-build and, more importantly, store images for multiple architectures. With this, we have the ability to store alternate images, which will enable us (future PR) to look at cross-building.

### CODE REVIEW

**Comment by deitch** [comment]
Fixed the tests to remove `-docker` flag, so everything can use cache. Now just to let CI run and see if it finds any issues, and fix them.

Definitely ready for a review.

**Comment by deitch** [comment]
Resolved all of those issues, just the packages tests still running.

**Comment by deitch** [comment]
CI is green

File: docs/packages.md [hunk 1]
```diff
@@ -67,6 +67,35 @@ Further, when building packages you need to be logged into hub with
...
+* the docker image cache
+
+If you want to build images and test and run them _in a standalone_ fashion locally, then you should pick the docker image cache. Otherwise, you should use the default linuxkit cache. LinuxKit defaults to building OS images using docker images from this cache,\
+only looking in the docker cache if instructed to via `linuxkit build --docker`.
+
+When using the linuxkit cache as the package build target, it creates all of the layers, the manifest that can be uploaded
+to a registry, and the multi-architecture index. If an image already exists for a different architecture in the cache,
+it updates the index to include additional manifests created.
+
+LinuxKit supports cross-building _operating system_ images, but does not (yet) support cross-building docker images.
```
Comment (justincormack) [inline docs/packages.md:84]: This statement is confusing, I think you need to explain it better

File: src/cmd/linuxkit/registry/manifest.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+const (
+	notaryServer                     = "https://notary.docker.io"
+	notaryDelegationPassphraseEnvVar = "NOTARY_DELEGATION_PASSPHRASE"
+	notaryAuthEnvVar                 = "NOTARY_AUTH"
+	dctEnvVar                        = "DOCKER_CONTENT_TRUST_REPOSITORY_PASSPHRASE"
+)
+
+var platforms = []string{
+	"linux/amd64", "linux/arm64", "linux/s390x",
```
Comment (justincormack) [inline src/cmd/linuxkit/registry/manifest.go:27]: I don't like hard coding this, don't have to change now but its a bit of a design smell.

**Comment by justincormack** [comment]
Generally design seems fine.

File: src/cmd/linuxkit/registry/manifest.go (hunk 2 above)
Comment (deitch) [reply src/cmd/linuxkit/registry/manifest.go:27]: I don't either, but it was there before. If we can come up with a good way, we can do a dedicated PR for it.

File: docs/packages.md (hunk 1 above)
Comment (deitch) [reply docs/packages.md:84]: Which one? The single sentence about what it can and cannot cross build? Or the line before about what it stores where?

**Comment by deitch** [comment]
I'll fix the doc comment you made, once I understand it better.

File: docs/packages.md (hunk 1 above)
Comment (deitch) [reply docs/packages.md]: Updated. Take a look @justincormack

**Review by justincormack — APPROVED** [review]

**Pull request merged**